ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node/Praos.hs

```diff
@@ -0,0 +1,312 @@
+        Comp $
+          BlockForging
+            { forgeLabel = label <> "_" <> shelleyBasedEraName (Proxy @era),
+              canBeLeader = canBeLeader,
+              updateForgeState = \_ curSlot _ ->
+                forgeStateUpdateInfoFromUpdateInfo
+                  <$> HotKey.evolve hotKey (slotToPeriod curSlot),
+              checkCanForge = \cfg curSlot _tickedChainDepState _isLeader ->
+                praosCheckCanForge
+                  (configConsensus cfg)
+                  curSlot,
+              forgeBlock = \cfg ->
+                forgeShelleyBlock
+                  hotKey
+                  canBeLeader
+                  cfg
+                  maxTxCapacityOverrides
+            }
+
+      startPeriod :: Absolute.KESPeriod
+      startPeriod = SL.ocertKESPeriod $ praosCanBeLeaderOpCert canBeLeader
+
+      slotToPeriod :: SlotNo -> Absolute.KESPeriod
+      slotToPeriod (SlotNo slot) =
+        SL.KESPeriod $ fromIntegral $ slot `div` praosSlotsPerKESPeriod
+
+{-------------------------------------------------------------------------------
+  ProtocolInfo
+-------------------------------------------------------------------------------}
+
+type BabbageEra c = AlonzoEra c
+
+-- | Parameters needed to run Babbage
+--
+--   Note that at the moment this links to Alonzo, since the Babbage era isn't
+--   around yet :-)
+data ProtocolParamsBabbage c = ProtocolParamsBabbage
+  { babbageProtVer :: SL.ProtVer,
+    babbageMaxTxCapacityOverrides :: TxLimits.Overrides (ShelleyBlock (Praos c) (BabbageEra c))
+  }
+
+protocolInfoPraosBabbage ::
+  forall m c.
+  ( IOLike m,
+    PraosCrypto c,
+    ShelleyCompatible (Praos c) (BabbageEra c),
+    TxLimits (ShelleyBlock (Praos c) (BabbageEra c))
+  ) =>
+  ProtocolParamsShelleyBased (BabbageEra c) ->
+  ProtocolParamsBabbage c ->
+  ProtocolInfo m (ShelleyBlock (Praos c) (BabbageEra c))
+protocolInfoPraosBabbage
+  protocolParamsShelleyBased
+  ProtocolParamsBabbage
+    { babbageProtVer = protVer,
+      babbageMaxTxCapacityOverrides = maxTxCapacityOverrides
+    } =
+    protocolInfoPraosShelleyBased
+      protocolParamsShelleyBased
+      (error "Babbage currently pretending to be Alonzo")
+      protVer
+      maxTxCapacityOverrides
+
+protocolInfoPraosShelleyBased ::
+  forall m era c.
+  ( IOLike m,
+    PraosCrypto c,
+    ShelleyCompatible (Praos c) era,
+    TxLimits (ShelleyBlock (Praos c) era),
+    c ~ EraCrypto era
+  ) =>
+  ProtocolParamsShelleyBased era ->
+  Core.TranslationContext era ->
+  SL.ProtVer ->
+  TxLimits.Overrides (ShelleyBlock (Praos c) era) ->
+  ProtocolInfo m (ShelleyBlock (Praos c) era)
+protocolInfoPraosShelleyBased
+  ProtocolParamsShelleyBased
+    { shelleyBasedGenesis = genesis,
+      shelleyBasedInitialNonce = initialNonce,
+      shelleyBasedLeaderCredentials = credentialss
+    }
+  transCtxt
+  protVer
+  maxTxCapacityOverrides =
+    -- TODO decide if we care about this. We're not running Babbage from gensis.
```

Comment:
a TODO to follow-up on

Reply:
`protocolInfoPraosShelleyBased` is only used for a chain that does not use the hard fork combinator. So, if this code is live, then it is running from Genesis.